src/clj/game/cards.clj

```diff
@@ -207,6 +207,69 @@
+                           (let [msg (str (join ", " (map #(let [{:keys [card number]} %
+                                                                 title (:title card)]
+                                                             (str number "[Credits] from " title))
+                                                          (vals selected-cards)))
+                                          (let [remainder (- target-count counter-count)]
+                                            (when (pos? remainder) (str " and " remainder "[Credits]"))))]
+                             (effect-completed state side (make-result eid {:number counter-count :msg msg}))))
+                         (effect-completed state side 0)))]
+     (if (or (not (pos? target-count))        ; there is a limit
+             (>= counter-count target-count)  ; paid everything
+             (zero? (count (provider-func)))) ; no more additional credit sources needed
+
+       {:effect pay-rest}
+
+       {:async true
+        :prompt (str "Select a credit providing card ("
+                     counter-count (when (and target-count (pos? target-count))
+                                     (str " of " target-count))
+                     " credits)")
+        :choices {:req #(and (in-coll? (map :cid (provider-func)) (:cid %))
+                             (case (-> % card-def :interactions :pay-credits :type)
+                               :recurring
+                               (pos? (get-counters % :recurring))
+
+                               :credit
+                               (pos? (get-counters % :credit))
+
+                               :custom
+                               true))}
+        :effect (req (let [pay-credits-type (-> target card-def :interactions :pay-credits :type)
+                           gained-credits (case pay-credits-type
+                                            :recurring
+                                            (or (add-prop state side target :rec-counter -1) 1)
+
+                                            :credit
+                                            (or (add-counter state side target :credit -1) 1)
+
+                                            :custom ;custom functions should return the number of credits provided
+                                            (or ((-> target card-def :interactions :pay-credits :custom) state side eid card nil) 1))
+                           selected-cards (update selected-cards (:cid target)
+                                                  ;; Store card reference and number of counters picked
+                                                  ;; Overwrite card reference each time
+                                                  #(assoc % :card target :number (inc (:number % 0))))
+                           counter-count (+ counter-count gained-credits)]
+                       (if (has-subtype? target "Stealth")
```

Comment:
As I said in a different comment, this will have to be moved out of this function and into pay-rest, cuz paying costs is a simultaneous thing: you either pay everything at the same time or pay nothing. Thus, you can't spend a credit from Cloak and a credit from Net Mercur to boost Houdini if Net Mercur doesn't already have a credit on it.

Reply:
That case is working right now though. When you use Houdini, `pick-credit-providing-cards` gets called. Now at this point here (line 264), we're waiting for the `:spent-stealth-credit` trigger, which causes the Net Mercur prompt to happen. You can now place a credit on Net Mercur and only afterwards go back to `pick-credit-providing-cards`, which now sees the Net Mercur credit. 

The downside of my implementation though is that paying for stuff is not simultaneous anymore. It is happening one credit at a time (until you press "Done" and `pay-rest` gets called).

Reply:
I just loaded it up, and it works incorrectly. I initiate a run on R&D, I have Net Mercur, Cloak, and Black Orchestra installed. I click Blorch, it gives me the credit selection prompt, so I click Cloak, and Net Mercur's prompt appears. I click "Gain 1 credit", and a credit appears on Net Mercur. I click on Net Mercur, and the credit comes off, and the rest of the cost is paid automatically.

> noah pays 1 from Cloak, 1 from Net Mercur and 1 to use Black Orchestra to add 2 strength and break up to 2 subroutines.
<img width="201" alt="image" src="https://user-images.githubusercontent.com/603677/58822685-144c9600-8606-11e9-8698-55fcd1635537.png">

Reply:
Oh, you're right of course. I was in too deep implementing stuff, that I forgot how the rules actually work :joy: